### PR TITLE
Add back not node

### PR DESCRIPTION
**Pull request by rui-mo**
Not node can be added back with the issue of NOT metadata filter fixed.

### CODE REVIEW

File: velox/dwio/common/MetadataFilter.cpp [hunk 1]
```diff
@@ -157,11 +157,17 @@ std::unique_ptr<MetadataFilter::Node> MetadataFilter::Node::fromExpression(
...
+      return nullptr;
+    }
+    return std::make_unique<NotNode>(std::move(negated));
+  }
   if (call->name() == "endswith" || call->name() == "contains" ||
       call->name() == "like" || call->name() == "startswith" ||
-      call->name() == "in" || call->name() == "rlike" ||
-      call->name() == "isnotnull" || call->name() == "coalesce" ||
-      call->name() == "might_contain") {
+      call->name() == "rlike" || call->name() == "isnotnull" ||
```
Comment (winningsix) [inline velox/dwio/common/MetadataFilter.cpp:169]: Question: why ignore ```in``` function?

Comment (rui-mo) [reply velox/dwio/common/MetadataFilter.cpp:169]: This is to fix a Velox UT. Based on that test, IN should be handled in Metadata filter.

**Comment by rui-mo** [comment]
Pass CI in https://github.com/oap-project/gluten/pull/1496.

**Comment by rui-mo** [comment]
The second commit in this PR is fetched from upstream.
cc @zhejiangxiaomai

**Comment by zhejiangxiaomai** [comment]
> The second commit in this PR is fetched from upstream. cc @zhejiangxiaomai

Thanks, I got it.

**Pull request merged**